### ISSUE TITLE
Add skater with adjustable mass to Intro screen?

### ISSUE THREAD

**Issue by arouinfar** [phetsims/energy-skate-park#263]
Kathy suggested adding a Custom skater with adjustable mass to the Intro screen. This would add quite a bit of flexibility, and may be desirable functionality for PhET-iO clients. 

When selecting "Custom", a NumberControl with a range of 1-100 kg would appear below the Skater ComboBox. We may need an additional piece of skater artwork for the Custom option, or we could consider using the default Skater 1. 

Let's discuss at design meeting.

**Comment by jessegreenberg** [comment]
My question for this is what would happen to control panel layout when the the NumberControl appears?

Would the panel grow and the measuring tape/timer toolbox panel shift down? 

Instead, could we have the NumberControl always visible and have its enabled state toggle?

Or could we use the mass control on the Playground screen for all screens?
![image](https://user-images.githubusercontent.com/6396244/80120435-fc9d3980-8558-11ea-9a8c-fefcad281660.png)

**Comment by jessegreenberg** [comment]
This was discussed in design meeting we are going to make some changes. @arouinfar is going to produce some new mock ups and update this issue.

**Comment by arouinfar** [comment]
4/23/20 design meeting notes

* Add Mass NumberControl to Intro screen (separated from Gravity by horizontal line)
* Below that, add "Skater" section with RadioButtonGroup of skater headshots. 
* Intro and Playground screens will have the same design.

This will take some mockups, so self-assigning.

**Comment by arouinfar** [comment]
I've mocked up the updated Skater controls. The skater headshot radio buttons are 30x30 px. We can fit 5 in a single row, though I don't think there's enough padding for this to be feasible on touch. If we move the radio buttons onto two rows, there'll be enough for all six skater options. We'll need to do a little bit of pixel polishing to buy the necessary vertical space, but I think it's doable if we reduce the height of the toolbox a bit and perhaps reduce the padding between the checkboxes. @ariel-phet what do you think?

![image](https://user-images.githubusercontent.com/8419308/80430675-8d06b180-88ac-11ea-8762-2f016127f010.png)

**Comment by ariel-phet** [comment]
@arouinfar I think I would just remove the word "Skater" and buy yourself some room. I think the radio button group is kind of self evident in what you are doing, and labeling it has no pedagogical value. Then I think the two rows of 3 would work well and not be too crowded

**Comment by arouinfar** [comment]
Great idea, @ariel-phet! If we remove "Skater" I think we'll have room for 35x35 px skater radio buttons. The toolbox height and inter-checkbox padding can remain untouched.

![image](https://user-images.githubusercontent.com/8419308/80512060-33e55f00-893a-11ea-859b-9d0b0683e7c9.png)

I've cleaned up the skater headshots a bit. Here are the assets @jessegreenberg:
[skaters.zip](https://github.com/phetsims/energy-skate-park/files/4547670/skaters.zip)

- [ ] Add Mass NumberControl to Intro screen (separated from Gravity by horizontal line)
- [ ] Below that, add RadioButtonGroup of skater headshots (right mockup above). Let's try 35x35 px radio buttons to start.
- [ ] Update Playground to match Intro layout.

**Comment by jessegreenberg** [comment]
@arouinfar @ariel-phet unfortunately this "grid" layout for buttons is not supported by RadioButtonGroup. See https://github.com/phetsims/sun/issues/513. vector-addition is using a custom solution to support this layout. If we want this layout in energy-skate-park it will take extra work and we will need to either
- Get to the bottom of https://github.com/phetsims/sun/issues/513, making general changes to RadioButtonGroup.
- Copy the custom work from vector-addition to get this working quickly in ESP.

I don't have time estimates for either approach yet, but wanted to let you know that this is not a straight forward change. How would you like to proceed?

**Comment by arouinfar** [comment]
Long term, I think we should absolutely address https://github.com/phetsims/sun/issues/513. Since this is the 2nd sim to need a grid layout, it seems like the appropriate time to generalize. However, I can't make that call. It's up to @ariel-phet to decide the next steps.

**Comment by ariel-phet** [comment]
@jessegreenberg just floating an idea...

It seems fixing up RadioButtonGroup would be a good project (perhaps post sim), it also seems that having an eye towards fixing it with a nice keyboard navigation strategy in mind and such would also be good.

If you were the one doing the RadioButtonGroup work, perhaps we could do the quick solution first, get ESP out the door, and this work would be your next "sim" task after ESP was done. We could then propogate the solution back to both Vector Addition and ESP.

Let me know your thoughts.

**Comment by jessegreenberg** [comment]
@ariel-phet that sounds resonable to me and I would be happy to take that on after energy-skate-park is republished.

**Comment by jessegreenberg** [comment]
This was done, the radio button group was added in the above commits. It looks great in the playground screen. On the intro screen, the panel doesn't have enough room because it also has the track radio buttons. @arouinfar what would you like to do about this?

**Comment by ariel-phet** [comment]
@jessegreenberg @arouinfar to my eye, the check box group at the top could be tightened up in terms of vertical spacing and actually look better and still be quite usable, so some pixels could be shaved off there.

**Comment by arouinfar** [comment]
@ariel-phet I was thinking along the same lines. However, we decided to add the Path checkbox to this screen in #176, so it will take a bit more polishing than reducing checkbox padding.

I was able to buy enough vertical space by 
* Reducing vertical padding between all elements in the panel to 10px.
* Reducing the height of the checkbox icons to 16 px.
* Reducing the height of the skater radio buttons to 30 px.
* Reducing the height of the toolbox to 50 px (icon size unchanged).

![image](https://user-images.githubusercontent.com/8419308/80740290-1a2c4f00-8ad5-11ea-8a60-2ba2c481cf4d.png)
